app/src/test/kotlin/fr/free/nrw/commons/logging/FileLoggingTreeUnitTest.kt

```diff
@@ -0,0 +1,143 @@
+package fr.free.nrw.commons.logging
+
+import android.util.Log
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.powermock.reflect.Whitebox
+import org.slf4j.Logger
+import java.lang.reflect.Method
+import java.util.concurrent.Executor
+
+class FileLoggingTreeUnitTest {
+
+    private lateinit var fileLoggingTree: FileLoggingTree
+
+    @Mock
+    private lateinit var executor: Executor
+
+    @Mock
+    private lateinit var logger: Logger
+
+    @Before
+    fun setUp() {
```

Comment:
I believe that all of the complex reflection-based method invocations would go away if you simply *planted a Timber tree* as part of the setup here, and used the public API through Timber.